resources/scripts/tap-to-junit-ext.sh

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+# license agreements. See the NOTICE file distributed with
+# this work for additional information regarding copyright
+# ownership. Elasticsearch B.V. licenses this file to you under
+# the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+set -eo pipefail
+
+if [ -n "${PIPELINE_LOG_LEVEL}" ] && [ "${PIPELINE_LOG_LEVEL}" == "DEBUG" ] ; then
+  set -x
+fi
+
+PATTERN=${1:?'Missing the pattern'}
+
+error=0
+for file in ${PATTERN} ; do
+echo "Parsing ${file}"
+filepath=$(dirname "$file")
+name=$(basename "$file")
+python3 -c "from itertools import count
+with open('${file}') as file:
+    for i in count():
+        firstline = next(file, None)
+        if firstline is None:
+            break
+        with open(f'${filepath}/{i}-${name}.ext', 'w') as out:
+            out.write(firstline)
+            for line in file:
+                out.write(line)
+                if line.startswith('running (cwd: ./'):
+                    break" || error=1
+done
+
+exit $error
```

Comment:
this is really an split script made in Python, it does not process the tap input to jUnit, the name is confusing. 
The bash part can be replaceable with a pure Python script it is weird to execute a Python script with more than one line in inline mode.

Reply:
I'll rename the file +1

Regarding the python script, I didn't want to create two different files, the shell script to iterate through the list of files for the given pattern and the python script itself.

What do you suggest?

Reply:
this script

```
#!/usr/bin/env python
# license agreements. See the NOTICE file distributed with
# this work for additional information regarding copyright
# ownership. Elasticsearch B.V. licenses this file to you under
# the Apache License, Version 2.0 (the "License"); you may
# not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     http:#www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing,
# software distributed under the License is distributed on an
# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
# KIND, either express or implied.  See the License for the
# specific language governing permissions and limitations
# under the License.

from itertools import count
import sys
import glob
import os.path

try:
    arg = sys.argv[1]
except IndexError:
    raise SystemExit(f"Usage: {sys.argv[0]} <file_pattern>")

pattern = sys.argv[1]

for filePath in glob.glob(pattern):
  print("Parsing  {}".format(filePath))
  dir = os.path.dirname(filePath)
  name = os.path.basename(filePath)

  with open(filePath) as file:
    for i in count():
        firstline = next(file, None)
        if firstline is None:
            break
        with open(f'{dir}/{i}-{name}.ext', 'w') as out:
            out.write(firstline)
            for line in file:
                out.write(line)
                if line.startswith('running (cwd: ./'):
                    break

```

Reply:
Gotcha!! That makes totally sense